templates/nextflow-ecs-cluster.yaml

```diff
@@ -0,0 +1,184 @@
+    Description: >
+      Specifies the Comma separated list of existing VPC Subnet
+      Ids where ECS instances will run.
+
+  AutoscalingGroupMaxSize:
+    Type: Number
+    Description: >
+      Specifies the number of instances to launch and register to the cluster.
+      Defaults to 1.
+    Default: '1'
+
+  SecurityIngressFromPort:
+    Type: Number
+    Description: >
+      Optional - Specifies the Start of Security Group port to open on
+      ECS instances - defaults to port 0
+    Default: '0'
+
+  SecurityIngressToPort:
+    Type: Number
+    Description: >
+      Optional - Specifies the End of Security Group port to open on ECS
+      instances - defaults to port 65535
+    Default: '65535'
+
+  SecurityIngressCidrIp:
+    Type: String
+    Description: >
+      Optional - Specifies the CIDR/IP range for Security Ports - defaults
+      to 0.0.0.0/0
+    AllowedPattern: ^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(\/([0-9]|[1-2][0-9]|3[0-2]))$
+    Default: '0.0.0.0/0'
+
+  RootEbsVolumeSize:
+    Type: Number
+    Description: Optional - Specifies the Size in GBs of the root EBS volume
+    Default: 30
+
+  EbsVolumeType:
+    Type: String
+    Description: Specifies the Type of (Amazon EBS) volume.
+    AllowedValues:
+      - io1
+      - gp2
+      - sc1
+      - st1
+      - standard
+    ConstraintDescription: Must be a valid EC2 volume type.
+    Default: 'gp2'
+
+  RootDeviceName:
+    Type: String
+    Description: Optional - Specifies the device mapping for the root EBS volume.
+    Default: '/dev/xvda'
+
+
+Resources:
+
+  EcsCluster:
+    Type: AWS::ECS::Cluster
+    Properties:
+      ClusterName: !Ref EcsClusterName
+      ClusterSettings:
+        - Name: containerInsights
+          Value: enabled
+
+  EcsRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+              - ec2.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Path: "/"
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceforEC2Role
+        - arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore
+
+  EcsInstanceProfile:
+    Type: AWS::IAM::InstanceProfile
+    Properties:
+      Roles:
+        - !Ref EcsRole
+
+  EcsSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: ECS Allowed Ports
+      VpcId: !Ref VpcId
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: !Ref SecurityIngressFromPort
+          ToPort: !Ref SecurityIngressToPort
+          CidrIp: !Ref SecurityIngressCidrIp
+
+  EcsLaunchConfiguration:
+    Type: AWS::AutoScaling::LaunchConfiguration
+    Properties:
+      ImageId: !Ref EcsAmiId
+      InstanceType: !Ref EcsInstanceType
+      AssociatePublicIpAddress: true
```

Comment:
hmm, i see you've passed in private subnets for `Subnet` however your associating public IPs?  this seems contradictory?

all instance are public?  don't we prefer them to be private for security reasons?

Reply:
Good catch -- this cluster is specifically for the NF tower software. We need to be able to access its frontend in a browser, so I'll change these to a single public subnet, unless you have a better solution.
All of the other clusters are created through NextFlow forge for running data jobs, and those will use exclusively private subnets.

Reply:
@zaro0508 Since you're out this week I am going to follow up with you in email